File: 1493-longest-subarray-of-1s-after-deleting-one-element/1493-longest-subarray-of-1s-after-deleting-one-element.py
class Solution:
    def longestSubarray(self, nums: List[int]) -> int:
        l , c, a, = 0,0,0
        for r in range(len(nums)):
            if nums[r] == 0:
                c += 1
            
            while c > 1:
                if nums[l] == 0:
                    c -=1
                l +=1
            a = max(a,r - l)
        return a

        

File: 1493-longest-subarray-of-1s-after-deleting-one-element/1493-longest-subarray-of-1s-after-deleting-one-element.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var longestSubarray = function(nums) {
    let count = 0, l = 0;
    let ans = 0;

    for(let r=0; r<nums.length; r++){
        count += nums[r] === 0 ? 1 : 0;

        while (count > 1){
            count -= nums[l] === 0 ? 1 : 0;
            l++;
        }

        ans = Math.max(ans , r - l + 1)
    }
    return ans - 1
};